powerauth-java-crypto/src/main/java/io/getlime/security/powerauth/crypto/lib/encryptor/ecies/model/EciesCryptogram.java

```diff
@@ -28,6 +28,28 @@ public class EciesCryptogram {
     private final byte[] mac;
     private final byte[] encryptedData;
     private final byte[] nonce;
+    private final byte[] associatedData;
+    private final byte[] timestamp;
+
+    /**
+     * Constructor for ECIES cryptogram used in encrypted requests.
+     *
+     * @param ephemeralPublicKey Ephemeral public key.
+     * @param mac MAC computed for key and data.
+     * @param encryptedData Encrypted data.
+     * @param nonce Nonce, required for protocol V3.1+.
+     * @param associatedData Associated data, required for protocol V3.2+.
+     * @param timestamp Timestamp, required for protocol V3.2+.
+     */
+    public EciesCryptogram(byte[] ephemeralPublicKey, byte[] mac, byte[] encryptedData, byte[] nonce,
```

Comment:
May we use a builder instead of this telescoping constructor with many parameters of the same type?

Reply:
Yes, we can use a builder.

Reply:
Thanks, I see the builder but you still want to keep the public constructor, right?

Reply:
Let's not worry about this, because we will refactor this class based on feedback from Petr.